`with pytest.warns(None)` was deprecated in pytest 7,
and removed in pytest 8

diff --git a/dask/dataframe/tests/test_accessors.py b/dask/dataframe/tests/test_accessors.py
index 0e68114..e4349c0 100644
--- a/dask/dataframe/tests/test_accessors.py
+++ b/dask/dataframe/tests/test_accessors.py
@@ -6,2 +6,3 @@ import numpy as np
 import pytest
+import warnings
 
@@ -104,19 +105,31 @@ def test_dt_accessor(df_ddf):
 
-    warning = FutureWarning if PANDAS_GE_210 else None
     # to_pydatetime returns a numpy array in pandas, but a Series in dask
     # pandas will start returning a Series with 3.0 as well
-    with pytest.warns(warning, match="will return a Series"):
-        ddf_result = ddf.dt_col.dt.to_pydatetime()
-    with pytest.warns(warning, match="will return a Series"):
-        pd_result = pd.Series(
-            df.dt_col.dt.to_pydatetime(), index=df.index, dtype=object
-        )
-    assert_eq(ddf_result, pd_result)
+    if PANDAS_GE_210:
+        with pytest.warns(FutureWarning, match="will return a Series"):
+            ddf_result = ddf.dt_col.dt.to_pydatetime()
+        with pytest.warns(FutureWarning, match="will return a Series"):
+            pd_result = pd.Series(
+                df.dt_col.dt.to_pydatetime(), index=df.index, dtype=object
+            )
+        assert_eq(ddf_result, pd_result)
 
-    assert set(ddf.dt_col.dt.date.dask) == set(ddf.dt_col.dt.date.dask)
-    with pytest.warns(warning, match="will return a Series"):
-        assert set(ddf.dt_col.dt.to_pydatetime().dask) == set(
-            ddf.dt_col.dt.to_pydatetime().dask
-        )
+        assert set(ddf.dt_col.dt.date.dask) == set(ddf.dt_col.dt.date.dask)
+        with pytest.warns(FutureWarning, match="will return a Series"):
+            assert set(ddf.dt_col.dt.to_pydatetime().dask) == set(
+                ddf.dt_col.dt.to_pydatetime().dask
+            )
+    else:
+        with warnings.catch_warnings():
+            warnings.simplefilter("error")
+            ddf_result = ddf.dt_col.dt.to_pydatetime()
+            pd_result = pd.Series(
+                df.dt_col.dt.to_pydatetime(), index=df.index, dtype=object
+            )
+            assert_eq(ddf_result, pd_result)
 
+            assert set(ddf.dt_col.dt.date.dask) == set(ddf.dt_col.dt.date.dask)
+            assert set(ddf.dt_col.dt.to_pydatetime().dask) == set(
+                ddf.dt_col.dt.to_pydatetime().dask
+            )
 
